cypress/page/registrationForm.page.ts

```diff
@@ -0,0 +1,64 @@
+type personalInformation = {
+    name: string
+    lastName: string
+    email: string
+    gender: string
+    mobileNumber: string
+    hobbies: string[]
+    currentAddress: string
+    state: string
+    city: string
+}
+class RestrationFormPage {
+    private formPageUrl: string
+    private firstName: string;
+    private lastName: string;
+    private email: string;
+    private mobileNumber: string;
+    private currentAdress: string
+    private state: string
+    private city: string
+
+    constructor() {
+        this.formPageUrl = "https://demoqa.com/automation-practice-form";
+        this.firstName = "#firstName";
+        this.lastName = "#lastName";
+        this.email = "#userEmail";
+        this.mobileNumber = "#userNumber";
+        this.currentAdress = "#currentAddress";
+        this.state = "#react-select-3-input";
+        this.city = "#react-select-4-input";
+    }
+    
+    public visitFormPage(): void {
+        cy.visit(this.formPageUrl);
+    }
+
+    public fillForm(userData : personalInformation): void {
+        cy.get(this.firstName).type(userData.name);
+        cy.get(this.lastName).type(userData.lastName);
+        cy.get(this.email).type(userData.email);
+        cy.contains(userData.gender).click()
+        cy.get(this.mobileNumber).type(userData.mobileNumber);
+        this.setHobbies(userData.hobbies);
+        cy.get(this.currentAdress).type(userData.currentAddress);
+        cy.get(this.state).focus().type(`${userData.state}{enter}`);
+        cy.get(this.city).focus().type(`${userData.city}{enter}`);
+        cy.get("#userForm").submit();
+    }
+
+    private setHobbies(userHobbies: string[]): void {
+        userHobbies.forEach(element => {
+            cy.contains(element).click();
+        });
+    }
+
+    public checkForm(userData : personalInformation): void {
+        Object.entries(userData).forEach((user) => {
+            user[0] === "hobbies" ? userData.hobbies.join(", ")  : cy.get('.modal').contains(user[1].toString())
+        });
+    }
```

Comment:
could you explain what you want to verify here, probably, we can think in another way to verify it since to me is hard to understand :)

Reply:
in this function i compare each registered field with the user information. i also use a ternary operator to convert the array of hobbies to a string and continue comparing.

Reply:
got it. However, I think that sometimes, writing more lines is better in terms of increasing the clarity and maintenance of our code. What about just write:
```
cy.get('.modal').contains(userData.name)
cy.get('.modal').contains(userData.lastName)
....
cy.get('.modal').contains(userData.hobbies.join(", "))
...
cy.get('.modal').contains(userData.city)
```

I know that is more code, but It will more easy to read. What do you think? 

Also we can ask to @manuelq12 what he thinks about since this is just my point of view :)

Reply:
@holgiosalos I totally agree with you and also I wouldn't recommend the use of a static index because in terms of maintenance that will make the test flaky if changes are applied.
I believe its a clever solution, I know this is a workshop and probably after you complete it you won't need to fix it in any matter but keep in mind

- Readability is a good practice for clean code
- Maintainability is essential for automated tests, we should always keep in mind this.

Reply:
@juan45202 I would leave it up to you if you feel comfortable with your solution or decide to apply Holmes changes, my suggestion is to apply Holmes changes as it will be easier for you to remember or someone elses to understand your code

Reply:
Thanks guys, if it is more clear the solution that you propose. I will change it.